extension-manifest-v3/src/background/utils/request.js

```diff
@@ -53,17 +53,17 @@ export default class ExtendedRequest extends Request {
     this.sourceHostname = data.sourceHostname;
   }
 
-  isFromOriginUrl(url) {
+  isFromOrigin(origin) {
     const { frameAncestors } = this._originalRequestDetails;
 
     /* Firefox APIs */
 
     if (frameAncestors && frameAncestors.length > 0) {
-      return url === frameAncestors[frameAncestors.length - 1].url;
+      return frameAncestors[frameAncestors.length - 1].url?.includes(origin);
     }
 
     if (this.sourceUrl) {
-      return url === this.sourceUrl;
+      return this.sourceUrl.includes(origin);
```

Comment:
Would `startsWith` be still correct (and more precise)?

Reply:
Sadly nope, I even gave an example in the description, which was a case where the bug was found. The sourceUrl is `https://www.onet.pl`, but the `url` from the original code was `http://onet.pl` (where we saved it as `onet.pl`). This is way we cannot compare full URLs as they are different (in case of Firefox, where it dispatch an navigation event even for the redirect).